test/module_test.cpp

```diff
@@ -277,6 +277,73 @@ TEST_CASE(parameter_name_order)
     EXPECT(param_names == names1);
 }
 
+TEST_CASE(insert_instructions_module)
```

Comment:
Even though we are implicitly calling insert_instructions() when using add_instructions(), the test name may be better as `add_instructions_module`. In other places in our source code, we directly call insert_instruction(). Maybe another example where we don't do the insert at the end of the module?

Reply:
Done